File: solutions/typescript/perfect-numbers/1/perfect-numbers.ts
export function classify(num: number): string {
  if (num < 1) {
    throw new Error("Classification is only possible for natural numbers.")
  }
  if (num === 1) {
    return "deficient"
  }
  var arr: number[] = [1]
  for (let i = 2; i <= num / 2; i++) {
    if (num % i === 0) {
      arr.push(i)
    }
  }
  var sum = arr.reduce((el, acc)=> acc + el, 0)
  if (sum === num) {
    return "perfect"
  } else if(sum < num) {
    return "deficient"
  } else {
    return "abundant"
  }
}

File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    divisor_sum = sum([x for x in range(2, number) if number % x == 0]) + 1
    if number == divisor_sum:
        return 'perfect'
    if number > divisor_sum:
        return 'abundant'
    else:
        return 'deficient'
